src/pages/profile/index.ts

```diff
@@ -37,9 +65,13 @@ export default class ProfilePage
         isConnected: this.connected,
         did: agent.profile.did,
         toggleConnection: () => (this.connected = !this.connected),
-        onProfileNameChange: (newName: string) => {
+        onProfileNameChange: async (newName: string) => {
           agent.profile.label = newName
           m.route.set('/:actor', {actor: agent.profile.label})
+          let contacts = ContactService.getContacts()
+          for(let contact of contacts) {
+            await this.sendProfile(contact)
+          }
```

Comment:
If we do as I suggest in my previous comment and move `sendProfile` messaging details to `agent.ts`, then that "frees up" the `sendProfile` method in this class to do the contact look up and then calling `agent.sendProfile` for each contact.